TransformationSystem/DB/TransformationDB.sql

```diff
@@ -35,6 +35,9 @@ CREATE TABLE Transformations (
     AgentType CHAR(32) DEFAULT 'Manual',
     Status  CHAR(32) DEFAULT 'New',
     FileMask VARCHAR(255),
+    InputMetaQuery VARCHAR(512) DEFAULT '',
```

Comment:
Instead of adding this field to the transformation, it would be more coherent to use the TransformationInputDataQuery table to store this information, and maybe use a flag to switch between using filters or the InputDataAgent. And add a table to store the OutputMetaQuery. This way you are also not as limited to 512 characters.
And is the FileMask parameter now totally redundant or is that still used for something?

Reply:
Indeed we could use a separate table to store the Input/OutputDataQuery, but for the InputDataQuery do you mean that we should use the InputDataQuery table as it's currently defined, with ParameterName, ParameterValue, ParameterType columns? Or do you agree to have a unique separate table storing Input and Output Metadata queries as VARCHAR fields?

Concerning the FileMask, it's totally redundant, so it should be removed.

Reply:
I think storing the input query as VARCHAR is not as flexible as having each metafield as a single entry. For once, because one is limited in the overall length, which is harder to control than the individual metadata names.
It would also cause some problems if the existing columns are changed.

Do you require some changes that cannot be accomodated by the separate columns?

Reply:
I was thinking for instance to the support of operators other than '='. It seems to me that currently this is the only one supported in the TransformationInputDataQuery, correct me if I'm wrong. Concerning eventual problems with the existing columns, you mean for backward compatibility? If this is your concern, maybe we could leave the TransformationInputDataQuery table as it is for usage with the 'InputDataAgent' flag enabled while using a different table for the 'filter' flag enabled? Concerning the lentgh of the field. I'm not expert about MySQL optimizations with respect to data access and storage requirements, so I would be happy to collect comments, for instance about the opportunity of using different types, like json columns, or blob...

Reply:
The TransformationInputDataQuery table also supports the `Dict` type, which should allow operators other than `=`
e.g.:
https://github.com/DIRACGrid/DIRAC/blob/5e4ab28334e7359fdb25c0962ad6f021acccdd38/TransformationSystem/DB/TransformationDB.py#L1010-L1012

Reply:
I've followed your suggestion and I've created a new table called:
TransformationMetaQueries with very similar structure of TransformationInputDataQuery.

The difference is that I've added a 'QueryType' field so that one can specify if the query is in 'Input' or an 'Output' meta query. In principle this table could replace the TransformationInputDataQuery, but for the moment I've leave it for backward compatibility so that one is able to use the 'old' way to specify InputDataQueries.